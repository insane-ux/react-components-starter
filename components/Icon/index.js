// @flow

import React from 'react'

const Icon = ({ className }: {
  className?: string,
}) => (
  <div className={className}>Icon</div>
)

Icon.defaultProps = {
  className: undefined,
}

export default Icon
