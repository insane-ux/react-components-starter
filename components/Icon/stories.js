// @flow

import React from 'react'
import { storiesOf } from '@kadira/storybook'
import Icon from './'

storiesOf('Icon', module)
  .add('default', () => (
    <Icon />
  ))
