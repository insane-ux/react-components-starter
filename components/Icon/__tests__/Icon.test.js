/* global it, expect, describe */

// @flow

import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import Icon from '../index'

describe('With Enzyme', () => {
  it('Icon with className', () => {
    const input = shallow(
      <Icon className="some" />,
    )

    expect(input.find('.some').prop('className')).toEqual('some')
  })
})

describe('With Snapshot Testing', () => {
  it('Icon with className', () => {
    const component = renderer.create(<Icon className="some" />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
