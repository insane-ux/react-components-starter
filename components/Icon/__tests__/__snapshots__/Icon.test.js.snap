// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Icon with className 1`] = `
<div
  className="some"
>
  Icon
</div>
`;
